examples/digital_fingerprinting/production/morpheus/dfp/stages/multi_file_source.py

```diff
@@ -0,0 +1,127 @@
+# Copyright (c) 2022, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import typing
+
+import fsspec
+import fsspec.utils
+import pandas as pd
+import srf
+
+from morpheus.config import Config
+from morpheus.io.deserializers import read_file_to_df
+from morpheus.pipeline.single_output_source import SingleOutputSource
+from morpheus.pipeline.stream_pair import StreamPair
+
+from ..utils.column_info import process_dataframe
+
+logger = logging.getLogger("morpheus.{}".format(__name__))
+
+
+class MultiFileSource(SingleOutputSource):
+    """
+    Source stage is used to load messages from a file and dumping the contents into the pipeline immediately. Useful for
+    testing performance and accuracy of a pipeline.
+
+    Parameters
+    ----------
+    c : `morpheus.config.Config`
+        Pipeline configuration instance.
+    filename : str
+        Name of the file from which the messages will be read.
+    iterative: boolean
+        Iterative mode will emit dataframes one at a time. Otherwise a list of dataframes is emitted. Iterative mode is
+        good for interleaving source stages.
+    file_type : `morpheus._lib.file_types.FileTypes`, default = 'auto'
+        Indicates what type of file to read. Specifying 'auto' will determine the file type from the extension.
+        Supported extensions: 'json', 'csv'
+    repeat: int, default = 1
+        Repeats the input dataset multiple times. Useful to extend small datasets for debugging.
+    filter_null: bool, default = True
+        Whether or not to filter rows with null 'data' column. Null values in the 'data' column can cause issues down
+        the line with processing. Setting this to True is recommended.
+    cudf_kwargs: dict, default=None
+        keyword args passed to underlying cuDF I/O function. See the cuDF documentation for `cudf.read_csv()` and
+        `cudf.read_json()` for the available options. With `file_type` == 'json', this defaults to ``{ "lines": True }``
+        and with `file_type` == 'csv', this defaults to ``{}``.
```

Comment:
Out of date docstrings.
I updated these in my PR if you want to copy/paste

Reply:
Done

Reply:
Done